iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<Configuration>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(GetAsync)} threw an exception: {ex}", nameof(GetAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Get configuration: {configurationId}", nameof(GetAsync));
+            }
+        }
+
+        /// <summary>
+        /// Retrieves specified number of configurations from every IoT hub partition.
+        /// Results are not ordered.
+        /// </summary>
+        /// <returns>The list of configurations.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task<IEnumerable<Configuration>> GetAsync(int maxCount, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Getting configuration: max count: {maxCount}", nameof(GetAsync));
+            try
+            {
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Get, GetConfigurationRequestUri(""), _credentialProvider, null, GetConfigurationsRequestUri(maxCount));
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<IEnumerable<Configuration>>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(GetAsync)} threw an exception: {ex}", nameof(GetAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Getting configuration: max count: {maxCount}", nameof(GetAsync));
+            }
+        }
+
+        /// <summary>
+        /// Update the mutable fields of the configuration registration
+        /// </summary>
+        /// <param name="configuration">The configuration object with updated fields.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object with updated ETags.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> UpdateAsync(Configuration configuration, CancellationToken cancellationToken = default)
```

Comment:
Azure SDK guidance suggests we use "Set" as the verb for methods like these instead of "Update"
```suggestion
        public virtual Task<Configuration> SetAsync(Configuration configuration, CancellationToken cancellationToken = default)
```